src/ripple/app/tests/Flow_test.cpp

```diff
@@ -709,6 +709,38 @@ struct Flow_test : public beast::unit_test::suite
             // found unfunded
             expect (!isOffer (env, bob, BTC (60), EUR (50)));
         }
+        {
+            // Do not produce more in the forward pass than the reverse pass
+            // This test uses a path that whose reverse pass will compute a
+            // 0.5 USD input required for a 1 EUR output. It sets a sendmax of
+            // 0.4 USD, so the payment engine will need to do a forward pass.
+            // Without limits, the 0.4 USD would produce 1000 EUR in the forward
+            // pass. This test checks that the payment procudes 1 EUR, as expected.
```

Comment:
Nice test.  And now I understand the motivation for the new code in BookStep.cpp

procudes -> produces?